CONTRIBUTING.rst

```diff
@@ -51,3 +51,61 @@ All interactions for this project are covered by the
 `PSF Code of Conduct <https://www.python.org/psf/codeofconduct/>`_. Everyone is
 expected to be open, considerate, and respectful of others no matter their
 position within the project.
+
+
+Run pre-commit linting locally
+------------------------------
+
+If you'd like, you can run the project's basic linting suite locally,
+either on-demand when you choose, or automatically against modified files
+whenever you commit your changes.
+
+They are also run in CI, so you don't have to run them locally, though doing
+so will help you catch and potentially fix common mistakes before pushing
+your changes and opening a pull request.
+
+This repository uses the `pre-commit <https://pre-commit.com/>`_ tool to
+install, configure and update a suite of hooks that check for
+common problems and issues, and fix many of them automatically.
+
+If your system has ``make`` installed, you can run the pre-commit checkers
+on the full repo by simply running ``make lint``. Note that this will
+install pre-commit in the current virtual environment if it isn't already,
+so make sure you've activated the environment you want it to use
+before running this command.
+
+Otherwise, you can install pre-commit by running ``pip install pre-commit``
+(or your choice of installer), and then run the hooks on all the files
+in the repo with ``pre-commit run --all-files``, or only on any files that
+have been modified but not yet committed with ``pre-commit run``.
+
+If you would like pre-commit to run automatically against any modified files
+every time you commit, install the hooks with ``pre-commit install``.
+Then, whenever you ``git commit``, pre-commit will run and report any issues
+it finds or changes it makes, and abort the commit to allow you to check,
+and if necessary correct them before committing again.
+
+
+Check and fix PEP spelling
+--------------------------
+
+To check for common spelling mistakes in your PEP and automatically suggest
+corrections, you can run the codespell tool through pre-commit as well.
+
+Like the linters, on a system with ``make`` available, it can be installed
+(in the currently-activated environment) and run on all files in the
+repository with a single command, ``make spellcheck``.
+
+For finer control or on other systems, after installing pre-commit as above
+(e.g. ``pip install pre-commit``), you can run it against only the files
+you've modified and not yet committed with
+``pre-commit run --hook-stage manual codespell``, or against all files with
+``pre-commit run --all-files --hook-stage manual codespell``.
+
+**Note**: While fixing spelling mistakes as part of more substantive
```

Comment:
Hmm, I guess we could, but after playing around with it I'm not sure it fits well as a full section, nor what a good title for it would be. We can just link the PEP spelling heading, no? Its not like such PRs have been _that_ frequent here...

Reply:
Sure, that'll be fine.

Reply:
Actually, after some discussion on #2350 , I think it would be a good idea to make this a top-level section to discuss what our general policy for changing older PEPs, what contributions are helpful, and which are likely to be declined, I'll modify it accordingly.

Reply:
Err, further revision—after I started drafting text for this section, given this is a broader policy than what was originally envisioned here and might require additional discussion, I think it would be best to keep this PR in scope and discuss this in a follow-up one. I also drafted some other updates for other out of date parts of the contributing guide and readme which can also be handled separately (in this case independently, since they don't conflict with either of these changes).